Chenyuan/EssayMonkey.py

```diff
@@ -0,0 +1,110 @@
+__author__ = 'Chenyuan'
+
+from random import randint
+
+#class EssayMonkey contains three methods: readFile, generateSentence, and essayMonkey.
+#readFile and generateSentence are internal methods called by essayMonkey.
+#This program is designed to use the default thesauruses: EssayMonkeyVerbs.txt, EssayMonkeyNouns.txt, and EssayMonkeyAdjectives.txt
+#In order to make the auto-generated sentences reasonable, a better approach is to use training sentences and machine learning algorithms to capture the relationship among words.
+class EssayMonkey(object):
+
+    #readFile method is used to read a comma delimited file and return a list of words.
+    #Run time complexity: big theta(1)
+    #Pre-condition: fName is a working file directory and the file is comma delimited.
+    #Post-condition: return a list of words
+    def readFile(this_object, fName):
+        words = list()
+        with open(fName) as f:
+            #read the file line by line
+            content = f.readlines()
+            for line in content:
+                #delete the newline at the end of the paragraph
+                line = line.replace("\n", "")
+                words = words + line.split(",")
+        return words
+
+    #generateSentence method is used generate and return one type of sentence.
+    #Run time complexity: big theta(1).
+    #Pre-condition: num is a integer and among 0 to 5.
+    #Post-condition: return one type of sentence.
+    def generateSentence(this_object, num):
+        sentence = None
+
+        #create a verb pool, a noun pool, and a adjective pool
+        verbs = this_object.readFile("EssayMonkeyVerbs.txt")
+        nouns =this_object. readFile("EssayMonkeyNouns.txt")
+        adjs =this_object. readFile("EssayMonkeyAdjectives.txt")
+
+        #Choose a random noun
+        noun = nouns[randint(0,len(nouns) - 1)]
+        #Choose two more random nouns
+        noun2 = nouns[randint(0,len(nouns) - 1)]
+        noun3 = nouns[randint(0,len(nouns) - 1)]
+        #Choose a random adjective
+        adj = adjs[randint(0, len(adjs) - 2)]
+        #Choose a random verb
+        #Since there is only one "shall" in the verb pool, the program won't be able to choose the tense of the verb based on if the index is odd or even.
+        verb = verbs[randint(0,len(verbs) - 1)]
+        if "/" in verb:
+            verb = verb.split("/")[randint(0,1)]
+
+
+        #Capitalize the first character of the noun
+        nounUpper = noun[0].upper() + noun[1:]
+        #Capitalize the first character of the adjective
+        adjUpper = adj[0].upper() + adj[1:]
+
```

Comment:
Instead of saving the value to a variable why not just call capitalize when you need to.